spring-kafka/src/main/java/org/springframework/kafka/support/serializer/JsonSerde.java

```diff
@@ -49,29 +50,43 @@
 	private final JsonDeserializer<T> jsonDeserializer;
 
 	public JsonSerde() {
-		this((ObjectMapper) null);
+		this(false);
+	}
+
+	public JsonSerde(boolean isKey) {
+		this((ObjectMapper) null, isKey);
 	}
 
 	public JsonSerde(Class<T> targetType) {
-		this(targetType, null);
+		this(targetType, null, false);
+	}
+
+	public JsonSerde(Class<T> targetType, boolean isKey) {
+		this(targetType, null, isKey);
 	}
 
 	public JsonSerde(ObjectMapper objectMapper) {
-		this(null, objectMapper);
+		this(null, objectMapper, false);
+	}
+
+	public JsonSerde(ObjectMapper objectMapper, boolean isKey) {
+		this(null, objectMapper, isKey);
 	}
 
 	@SuppressWarnings("unchecked")
-	public JsonSerde(Class<T> targetType, ObjectMapper objectMapper) {
+	public JsonSerde(Class<T> targetType, ObjectMapper objectMapper, boolean isKey) {
```

Comment:
I think we are loosing here ctor for backward compatibility.

You know, having such a complexity I'd prefer to come to the setter idea. The setter can be like this:
```
public JsonSerde setUseTypeMapperForKey(boolean)
```
Therefore we still can use it as a fluent API for a sample you've shown before.

Reply:
Ok sounds good!